python/dgl/nn/pytorch/explain/pgexplainer.py

```diff
@@ -0,0 +1,356 @@
+        training : bool
+            Training the explanation network.
+        kwargs : dict
+            Additional arguments passed to the GNN model.
+
+        Returns
+        -------
+        Tensor
+            Classification probabilities given the masked graph. It is a tensor of
+            shape :math:`(1, L)`, where :math:`L` is the different types of label
+            in the dataset.
+        Tensor
+            Edge weights which is a tensor of shape :math:`(E)`, where :math:`E`
+            is the number of edges in the graph. A higher weight suggests a larger
+            contribution of the edge.
+
+        Examples
+        --------
+
+        >>> import torch as th
+        >>> import torch.nn as nn
+        >>> import dgl
+        >>> from dgl.data import GINDataset
+        >>> from dgl.dataloading import GraphDataLoader
+        >>> from dgl.nn import GraphConv, PGExplainer
+        >>> import numpy as np
+
+        >>> # Define the model
+        >>> class Model(nn.Module):
+        ...     def __init__(self, in_feats, out_feats):
+        ...         super().__init__()
+        ...         self.conv = GraphConv(in_feats, out_feats)
+        ...         self.fc = nn.Linear(out_feats, 2)
+        ...         nn.init.xavier_uniform_(self.fc.weight)
+        ...
+        ...     def forward(self, g, h, embed=False, edge_weight=None):
+        ...         h = self.conv(g, h, edge_weight=edge_weight)
+        ...         if not embed:
+        ...             g.ndata['h'] = h
+        ...             hg = dgl.mean_nodes(g, 'h')
+        ...             return self.fc(hg)
+        ...         else:
+        ...             return h
+
+        >>> # Load dataset
+        >>> data = GINDataset('MUTAG', self_loop=True)
+        >>> dataloader = GraphDataLoader(data, batch_size=64, shuffle=True)
+
+        >>> # Train the model
+        >>> feat_size = data[0][0].ndata['attr'].shape[1]
+        >>> model = Model(feat_size, data.gclasses)
+        >>> criterion = nn.CrossEntropyLoss()
+        >>> optimizer = th.optim.Adam(model.parameters(), lr=1e-2)
+        >>> for bg, labels in dataloader:
+        ...     preds = model(bg, bg.ndata['attr'])
+        ...     loss = criterion(preds.squeeze(1).float(), labels.float())
+        ...     optimizer.zero_grad()
+        ...     loss.backward()
+        ...     optimizer.step()
+
+        >>> # Initialize the explainer
+        >>> explainer = PGExplainer(model, data.gclasses)
+
+        >>> # Train the explainer
+        >>> # Define explainer temperature parameter
+        >>> init_tmp, final_tmp = 5.0, 1.0
+        >>> optimizer_exp = torch.optim.Adam(explainer.parameters(), lr=0.01)
+        >>> for epoch in range(20):
+        ...     tmp = float(init_tmp * np.power(final_tmp / init_tmp, epoch / 20))
+        ...     for bg, labels in dataloader:
+        ...          loss = explainer.train_step(bg, bg.ndata['attr'], tmp)
```

Comment:
The example here actually assumes the support for a batch of graphs.

Reply:
Yes, because I thought using batch graphs would be more common when someone is using pgexplainer.

Reply:
Then you need to cover that in the parameter/returns description. Previously you assumed a single graph there.

Reply:
Doesn't a dgl batch combine batches of graphs into a single graph?  https://docs.dgl.ai/en/1.0.x/generated/dgl.batch.html

Reply:
`train_step` documentation updated.

Reply:
> Doesn't a dgl batch combine batches of graphs into a single graph? https://docs.dgl.ai/en/1.0.x/generated/dgl.batch.html

Yes, but a readout function like `dgl.mean_nodes` will yield a batch dimension.